packages/react-core/src/components/Chip/__tests__/Chip.test.tsx

```diff
@@ -1,59 +1,234 @@
+});
+
+test(`Renders concatenated aria-label on action close button by default`, () => {
+  render(<Chip>Chip text</Chip>);
+
+  expect(screen.getByRole('button')).toHaveAccessibleName('close Chip text');
+});
+
+test(`Renders custom aria-label on action close button when closeBtnAriaLabel is passed`, () => {
+  render(<Chip closeBtnAriaLabel="custom label">Chip text</Chip>);
+
+  expect(screen.getByRole('button')).toHaveAccessibleName('custom label Chip text');
+});
+
+test(`Does not render close button action when isOverflowChip is true`, () => {
+  render(<Chip isOverflowChip>Chip text</Chip>);
+
+  // Because overflow chip renders as a button, we need to add the accessible name to the query
+  expect(screen.queryByRole('button', { name: 'close Chip text' })).not.toBeInTheDocument();
+});
+
+test(`Does not render close button action when isReadOnly is true`, () => {
+  render(<Chip isReadOnly>Chip text</Chip>);
+
+  expect(screen.queryByRole('button')).not.toBeInTheDocument();
+});
+
+test(`Does not render with class ${styles.modifiers.overflow} when isOverflowChip is not passed`, () => {
+  render(<Chip data-testid="container">Chip text</Chip>);
+
+  expect(screen.getByTestId('container')).not.toHaveClass(styles.modifiers.overflow);
+});
+
+test(`Renders with class ${styles.modifiers.overflow} when isOverflowChip is true`, () => {
+  render(
+    <Chip isOverflowChip data-testid="container">
+      Chip text
+    </Chip>
+  );
+
+  expect(screen.getByTestId('container')).toHaveClass(styles.modifiers.overflow);
+});
+
+test(`Does not render with type attribute when isOverflowChip is true and component is not button`, () => {
+  render(
+    <Chip component="span" isOverflowChip data-testid="container">
+      Chip text
+    </Chip>
+  );
+
+  expect(screen.getByTestId('container')).not.toHaveAttribute('type');
+});
+
+test(`Renders with type attribute when isOverflowChip is true and component is button`, () => {
+  render(
+    <Chip component="button" isOverflowChip data-testid="container">
+      Chip text
+    </Chip>
+  );
+
+  expect(screen.getByTestId('container')).toHaveAttribute('type', 'button');
+});
+
+test(`Calls onClick when close button action is clicked for default chip`, async () => {
+  const user = userEvent.setup();
+  const onClickMock = jest.fn();
+
+  render(<Chip onClick={onClickMock}>Chip text</Chip>);
+
+  await user.click(screen.getByRole('button', { name: 'close Chip text' }));
+
+  expect(onClickMock).toHaveBeenCalledTimes(1);
+});
+
+test(`Calls onClick when chip is clicked for overflow chip`, async () => {
+  const user = userEvent.setup();
+  const onClickMock = jest.fn();
+
+  render(
+    <Chip isOverflowChip component="button" onClick={onClickMock}>
+      Chip text
+    </Chip>
+  );
+
+  await user.click(screen.getByRole('button', { name: 'Chip text' }));
+
+  expect(onClickMock).toHaveBeenCalledTimes(1);
+});
+
+test('Matches snapshot', () => {
+  const { asFragment } = render(<Chip>Chip text</Chip>);
```

Comment:
I think we probably should add a fixed `id` and `ouiaId` to the snapshot so that we don't need to worry about the snapshots changing just because more tests get added or what have you.

Reply:
Looks like it still needs an `ouiaId`? Or am I missing something.